OpenSim/Simulation/GeodesicWrap/GeodesicPathSolver.h

```diff
@@ -0,0 +1,45 @@
+#ifndef OPENSIM_GEODESIC_PATH_SOLVER_H_
+#define OPENSIM_GEODESIC_PATH_SOLVER_H_
+
+// INCLUDE
+
+#include "OpenSim/Simulation/GeodesicWrap/GeodesicCurveState.h"
+#include "GeodesicWrapObject.h"
+
+namespace OpenSim {
+
+class GeodesicPathSolver
+{
+
+    public:
+
+    GeodesicPathSolver() = default;
+
+    struct GeodesicWrappingPath {
+        // Useful fields...
+        double length = NAN;
+        double lengtheningSpeed = NAN;
+        // Log of points
+    };
+
+    // 1. Call shooter on all surface objects
+    // 2. Compute wrapping-error (stop here if done)
+    // 2. Compute GeodesicVariation from the surface geodesics
+    // 3. Apply variation to local surface state.
+    // 4. Repeat until error converges.
+    void calcWrappingPath(
```

Comment:
If the `GeodesicWrapObject`s store a pointer to the surface `Frame`s, then we'll need to pass the model's `SimTK::State` as an argument here.